### ISSUE TITLE
Attempt to make `AsyncSessionHandle` satisfy `Send`

### ISSUE THREAD

**Issue by mortendahl** [tf-encrypted/moose#912]
It currently seems impossible to launch new sessions in different threads/tasks since `AsyncSessionHandle` is not `Send`. The reason for this is the `RwLockWriteGuard` that `join_on_first_error` takes.

This issue is about removing that limitation, making it possible to launch and/or wait on session in different tasks.

While it can be done in a follow-up issue, Rudolph should also be updated to do so when a solution has been found.
